subsys/mgmt/fmfu/src/mgmt_fmfu.c

```diff
@@ -0,0 +1,281 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <zephyr.h>
+#include <logging/log.h>
+#include <mgmt/mgmt.h>
+#include <nrf_fmfu.h>
+#include <mgmt/mgmt_fmfu.h>
+#include "cborattr/cborattr.h"
+#include <stats/stats.h>
+
+LOG_MODULE_REGISTER(mgmt_mfu, CONFIG_MGMT_MFU_LOG_LEVEL);
+
+/* Define full modem update mgmt spesific SMP return codes */
+#define MGMT_ERR_EMODEM_INVALID_COMMAND 200
+#define MGMT_ERR_EMODEM_FAULT           201
+
+/* Define full modem mgmt commands */
+#define FMFU_MGMT_ID_GET_HASH 0
+#define FMFU_MGMT_ID_UPLOAD 1
+
+/* Define full modem IPC failures*/
+#define NRF_FMFU_RET_COMMAND_FAILED -3
+#define NRF_FMFU_RET_COMMAND_FAULT -4
+
+/* Struct used for parsing upload requests. */
+struct firmware_packet {
+	uint8_t *data_ptr;
+	uint8_t data[SMP_PACKET_MAXIMUM_TRANSMISSION_UNIT];
+	uint32_t target_address;
+	uint32_t data_len;
+};
+
+static int mgmt_get_memory_hash(struct mgmt_ctxt *ctxt);
```

Comment:
Remove these forward declarations and declare the variable which needs them further down. At least see if it looks better, IMO the fmfu handler registration is not something that has to be in the top of the file.